### ISSUE TITLE
Crash when adding new track after reopening file

### ISSUE THREAD

**Issue by marksolly** [mltframework/shotcut#258]
Probably related to https://github.com/mltframework/shotcut/issues/257

Version: 16.06.08
OS: Windows 7 64 Bit

Steps to reproduce:
1. Open shotcut
2. Add two video tracks with video
3. Save the project
4. Close & reopen shotcut
5. Try to add a new video track
5. Crash

**Comment by marksolly** [comment]
Possibly a regression of https://github.com/mltframework/shotcut/issues/149

**Comment by ddennedy** [comment]
Unable to reproduce.

**Comment by marksolly** [comment]
Is there some other information I can provide to aid in reproduction?
A screen cast?
Logs?
Can you point me at the correct tool set to produce a stack trace?

**Comment by ddennedy** [comment]
A screencast would be good perhaps to reveal something missing, but I need to be able to reproduce it.

The tool to generate a stack trace is gdb, but I do not know how easy it is to get that and use it on Windows. I mean I get it through a much larger suite of development tools. If you have it and know what to do, great. Otherwise, I do not volunteer to guide you through that.

**Comment by marksolly** [comment]
I've recorded a screen cast showing steps to reproduce this bug and also #257 
View here: https://www.dropbox.com/s/kcvstoowcv9mil8/Shotcut%20Issues%20257%20and%20258%20.avi?dl=0

I note that the crash occurs even if I do not add an opacity filter as shown in the video.

I'll try to run gdb next and capture a stack trace.

**Comment by marksolly** [comment]
I was not able to get any build of gdb for windows to work but I did get a stack trace from WinDbg. If you can point me at the debug symbol files for windows I might be able to fill in some of those hex addresses with method names.

I also saved a dump file but that's 135Mb zipped so I'll only upload that if you want it.

```
(1190.1194): Access violation - code c0000005
Child-SP          RetAddr           Call Site
00000000`00227e30 00000000`69ad712b libmlt_6!mlt_service_connect_producer+0xc
00000000`00227e90 00000000`69ad75a7 libmlt_6!mlt_transition_connect+0x1b
00000000`00227ee0 00000000`6400abcd libmlt_6!mlt_field_plant_transition+0x17
00000000`00227f20 00000000`004a2318 libmlt___3!ZN3Mlt7Tractor16plant_transitionERNS_10TransitionEii+0x6b
00000000`00227f60 00000000`004b19c9 shotcut+0xa2318
00000000`002280b0 00000000`00ab9845 shotcut+0xb19c9
00000000`00228130 00000000`004ec010 Qt5Widgets!ZN10QUndoStack4pushEP12QUndoCommand+0x35
00000000`002281b0 00000000`004ec9bc shotcut+0xec010
00000000`00228270 00000000`66231dba shotcut+0xec9bc
00000000`002282c0 00000000`661ca835 Qt5Qml!ZN14QQmlMetaObject34resolveGadgetMethodOrPropertyIndexEN11QMetaObject4CallEPPKS0_Pi+0x16a
00000000`00228320 00000000`661cb7ca Qt5Qml!ZN3QV414QObjectWrapper4wrapEPNS_15ExecutionEngineEP7QObject+0x1805
00000000`00228600 00000000`661cc042 Qt5Qml!ZN3QV414QObjectWrapper4wrapEPNS_15ExecutionEngineEP7QObject+0x279a
00000000`00228700 00000000`661dd17d Qt5Qml!ZN3QV413QObjectMethod12callInternalEPNS_8CallDataE+0x4a2
00000000`00228910 00000000`0ea00cc5 Qt5Qml!ZN3QV47Runtime12callPropertyEPNS_15ExecutionEngineEiPNS_8CallDataE+0x23d
00000000`002289e0 00000000`0471c680 0xea00cc5
00000000`002289e8 00000000`00228a20 0x471c680
00000000`002289f0 00000000`68a44a9f 0x228a20
00000000`002289f8 00000000`61976e82 Qt5Core!ZN8QVariant6createEiPKv+0x4f
00000000`00228a00 00000000`0db46180 Qt5Gui!ZN22QInputMethodQueryEvent8setValueEN2Qt16InputMethodQueryERK8QVariant+0x82
00000000`00228a70 00000000`06c4ace0 0xdb46180
```

**Comment by ddennedy** [comment]
The video adds little to your already good description. In other words, it did not reveal a misunderstanding, but sadly I still cannot reproduce it. The fact that 'C' does not toggle clearly indicates the track compositor component could not be located, and surely it crashes for the same reason. Yet, your borked.mlt does show them in place (frei0r.cairoblend).This does help to narrow down the problem, but not sure what exactly is the problem - perhaps some memory corruption or race condition.

If you install mingw-w64, you will get gdb. The debug symbols from gcc are incompatible with WinDbg, and the dump file is useless to me. If you are a developer, [you can help](https://www.shotcut.org/notes/windowsdev/), and I can give you the pre-built SDK to bootstrap the process. You do not need to use Qt Creator, but it is highly recommended and  mingw and gcc are mandatory.

**Comment by marksolly** [comment]
I am a developer. Although gnu c++ is not my usual language/toolchain, I'm
prepared to try it.

Does your SDK include a build of gdb? I did install mingw-w64 from a couple
of sources but both refused to attach to shotcut.exe. I'll study the
documentation some more.

**Comment by ddennedy** [comment]
The SDK is here:
http://builds.us.meltytech.s3.amazonaws.com/shotcut/shotcut-win64-sdk-160619.zip

**Comment by marksolly** [comment]
This issue has been resolved in Shotcut v16.07.01. Origin remains unknown.

**Comment by ddennedy** [comment]
Thanks for following up on this, Mark.
